kubernetes/e2e_test/test_client.py

```diff
@@ -105,11 +106,13 @@ def test_pod_apis(self):
         self.assertFalse(resp.peek_stdout())
         self.assertEqual("test string 2", line)
         resp.write_stdin("exit\n")
-        resp.update(timeout=5)
+        select.select([resp.sock.sock], [], [], timeout=5)
+        resp.update()
         line = resp.read_channel(ERROR_CHANNEL)
         status = json.loads(line)
         self.assertEqual(status['status'], 'Success')
-        resp.update(timeout=5)
+        select.select([resp.sock.sock], [], [], timeout=5)
+        resp.update()
```

Comment:
I'm not sure if we want to ask existing users to do this. On the plus side this would be a hard break (unknown parameter) instead of some undefined behavior :(

cc @yliaog

Reply:
yes, that's a very good point: I was really just trying to remove the select call from the update method for sake of #602 and https://github.com/kubernetes-client/python-base/issues/106

okay, let me close this one and try the solution you proposed in the base PR